Lib/test/test_zipfile/test_core.py

```diff
@@ -3447,6 +3447,53 @@ def test_too_short(self):
         self.assertEqual(
             b"zzz", zipfile._Extra.strip(b"zzz", (self.ZIP64_EXTRA,)))
 
+class StoredZipExtFileRandomAccessTest(unittest.TestCase):
+    def test_random_access(self):
+        from _pyio import BytesIO
+        class StatIO(BytesIO):
+            def __init__(self):
+                super().__init__()
+                self.bytes_read = 0
+
+            def read(self, size=-1):
+                bs = super().read(size)
+                self.bytes_read += len(bs)
+                return bs
+
+            def get_bytes_read(self):
+                return self.bytes_read
+
+        sio = StatIO()
+        # 100000 bytes
+        txt = b'0123456789'*10000
```

Comment:
```suggestion
        txt = b'0123456789' * 100000
```

Do you really need that many bytes? can't you just live with 20k? In addition, because the data is uniform and the jumps are all multiples of 10, you have a period. I advise you to use different jumps and read sizes just to check that the data you read is really the one at the correct place.